src/MainWindow.vala

```diff
@@ -154,6 +152,23 @@ public class Maps.MainWindow : Adw.ApplicationWindow {
             map_widget.go_to_current ();
         });
 
+
+        var search_key_controller = new Gtk.EventControllerKey ();
+        search_key_controller.key_pressed.connect ((keyval, keycode, state) => {
+            switch (keyval) {
+                // Intercept space key so it's not used for list activation: https://github.com/elementary/maps/issues/150
+                case Gdk.Key.KP_Space:
+                case Gdk.Key.space:
+                    search_key_controller.forward (search_entry.get_delegate ());
+                    return Gdk.EVENT_STOP;
+            }
```

Comment:
```suggestion
                    return Gdk.EVENT_STOP;
                default:
                    // NOP
                    break;
            }
```

I agree with you to use switch-case here, but I prefer to explicit that it's intentional that we don't handle other keys.